iothub/device/src/Pipeline/TransportHandlerFactory.cs

```diff
@@ -12,43 +11,31 @@ internal class TransportHandlerFactory : ITransportHandlerFactory
     {
         public IDelegatingHandler Create(PipelineContext context)
         {
-            ClientConfiguration clientConfiguration = context.ClientConfiguration;
-            Func<MethodRequestInternal, Task> onMethodCallback = context.MethodCallback;
-            Action<TwinCollection> onDesiredStatePatchReceived = context.DesiredPropertyUpdateCallback;
-            Func<string, Message, Task> onModuleEventReceivedCallback = context.ModuleEventCallback;
-            Func<Message, Task> onDeviceMessageReceivedCallback = context.DeviceEventCallback;
+            IotHubClientTransportSettings transportSettings = context.IotHubClientTransportSettings;
 
-            if (clientConfiguration.ClientOptions.TransportSettings is IotHubClientAmqpSettings iotHubClientAmqpSettings)
+            if (transportSettings is IotHubClientAmqpSettings iotHubClientAmqpSettings)
             {
                 return new AmqpTransportHandler(
                     context,
-                    iotHubClientAmqpSettings,
-                    onMethodCallback,
```

Comment:
These callbacks are a part of the pipeline context which is passed into the transport layer. It is not required for us to extract these here and pass them in again.